source/StellarObject.cpp

```diff
@@ -155,6 +156,18 @@ double StellarObject::Distance() const
 
 
 
+// Change the distances an object is visible from, based on ship attributes.
+void StellarObject::UpdateDistanceVisibility(const Ship *flagship)
+{
+	if(!flagship || trueDistanceInvisible)
```

Comment:
```suggestion
	if(!flagship || !trueDistanceInvisible)
```
In the current version, this would not set the distances if they were defined in the data.

Actually, does this function have a reason to exist in the current implementation?

Reply:
Can you elaborate on what kind of bug you found?

Reply:
Without this, the bodies were always visible.

Reply:
Oh wait I see nevermind